src/software/simulation/physics/physics_field.cpp

```diff
@@ -5,6 +5,11 @@
 PhysicsField::PhysicsField(std::shared_ptr<b2World> world, const Field &field)
     : field(field)
 {
+    if (!field.isValid())
```

Comment:
1. fyi that this is being removed in #1413
2. the  exception should be reflected in the header

Reply:
added the exception to the header